internal/controller/pipelinerollout_ppnd.go

```diff
@@ -53,7 +54,7 @@ func (r *PipelineRolloutReconciler) processExistingPipelineWithPPND(ctx context.
 	// if it's safe to Update and we need to, do it now
 	if pipelineNeedsToUpdate {
 		pipelineRollout.Status.MarkPending()
-		if !shouldBePaused || (shouldBePaused && isPipelinePausedOrUnpausible(ctx, existingPipelineDef)) {
+		if !shouldBePaused || (shouldBePaused && checkPipelineStatus(ctx, existingPipelineDef, numaflowv1.PipelinePhasePaused)) {
```

Comment:
I actually don't need to check for the case of "won't pause/unpausible" here as it has already been checked in the derivation of `shouldBePaused`, which will be set to false here for the case of "won't pause/unpausible".